sources/ActionView.cpp

```diff
@@ -29,9 +30,30 @@ ActionView::ActionView(const char* name, BMessage* action, const int32& flags)
 	fValueBox = new AutoTextControl("valuebox", NULL, NULL, new BMessage());
 	fValueBox->SetDivider(0);
 
+	fAddButton = new BButton(kAdd);
+	fRemoveButton = new BButton(kRemove);
+
+	BMessage* msg = new BMessage(MSG_ADD_ACTION);
+	msg->AddPointer(kAdd, this);
+	fAddButton->SetMessage(msg);
+
+	msg = new BMessage(MSG_REMOVE_ACTION);
+	msg->AddPointer(kRemove, this);
+	fRemoveButton->SetMessage(msg);
+
+	BFont font;
+	GetFont(&font);
+	float width = font.StringWidth("+-") * 2;
+	BSize size(width, width);
+
+	fAddButton->SetExplicitSize(size);
+	fRemoveButton->SetExplicitSize(size);
+
 	BLayoutBuilder::Group<>(this, B_HORIZONTAL, B_USE_DEFAULT_SPACING)
 		.Add(fActionField, 0)
 		.Add(fValueBox)
+		.Add(fRemoveButton, 0)
+		.Add(fAddButton, 0)
```

Comment:
Yes HALF_ITEM tightens things up nicely.

I also realized, that there's a spacing issue if the action has only one pop-up menu, i.e. no text control, like "Delete". Add it as a second action and see that there's now a bigger gap in front of the +/- buttons.

This solves it:

```
	BLayoutBuilder::Group<>(this, B_HORIZONTAL, B_USE_HALF_ITEM_SPACING)
		.Add(fActionField, 0)
		.Add(fValueBox)
		.AddGroup(B_HORIZONTAL, 0)
			.Add(fRemoveButton)
			.Add(fAddButton)
			.End()
		.End();
```